src/keystore/InMemoryKeystore.ts

```diff
@@ -272,9 +277,43 @@ export default class InMemoryKeystore implements Keystore {
           'missing recipient'
         )
       }
-      const invitation = InvitationV1.createRandom(req.context)
+      // const invitation = InvitationV1.createRandom(req.context)
       const created = nsToDate(req.createdNs)
       const recipient = toSignedPublicKeyBundle(req.recipient)
+
+      const secret = await this.v2Keys.sharedSecret(
+        recipient,
+        this.v2Keys.getCurrentPreKey().publicKey,
+        false
+      )
+
+      const msg = new TextEncoder().encode(
+        JSON.stringify({
+          conversationId: req.context?.conversationId || '',
+          participants: [
+            this.accountAddress,
+            await recipient.walletSignatureAddress(),
+          ].sort(),
+        })
+      )
+
+      const topic = sha256(
+        await hmacSha256Sign(Buffer.from(secret), Buffer.from(msg))
+      )
+
+      const salt = getRandomValues(new Uint8Array(KDFSaltSize))
```

Comment:
Hrmmm. If we have random values in the salt, then repeated invites are going to be different. @jazzz thoughts on what we should use as a salt value here?

Reply:
Does it matter if the invite is different if the topic is still the same?

Reply:
Unfortunately so, since we are only going to store one key per topic in the `InviteStore`. And if we had more than one we would need to somehow choose the right one for each message.

Reply:
The salt is optional in HKDF, but I'm not sure what the practical implications are to omit

Reply:
Yeah this is problematic for the same reason using AES-GCM encrypt to derive `topicKey` [was a problem](https://www.notion.so/RAPID-Dedupe-conversations-via-Determinism-18735a19ed1140fcbf95c2ea801c5422?d=f4e661a6b1d84baf83e8427dc97a79af#9fc9b372d558435cb7f7f31c779b20c9). Any randomness introduced will not be replicated by an independent client making an invite to the same recipient at the same time.

We could deterministically derive the salt as `hmac(key=secret, fixed_string=...)` but I need a second pair of eyes on that scheme.

Reply:
Salt is optional: https://crypto.stackexchange.com/a/64154 - we should be fine with a fixed salt too. The Diffie-hellman output should provide sufficient entropy when passed through the `hkdf`.

Reply:
Added a salt constant in https://github.com/xmtp/xmtp-js/pull/326/commits/6632dba65169b2da3f3d384abd29b895cdb481c2.

Reply:
In this case the salt should be static for domain separation, and randomness should be supplied via the info field if necessary. 

The guarantees we require from HKDF in this application don't require this, but I would suggest for completeness that the info field be set using "|".join(["0"] + addr_strings.sort()).

info="0|0x744f31bA5e8F6bAE5287fCc5c34476A2267bCaA2|0xc86aE9599A2bB5a99b19D2b8AD1BF89082023d63"

The Prefixed sequence number will allow derivation of future iterations if needed, while the Additions on the addresses adds entropy to the context.

Salt constant looks great.